apollo-router/src/router_factory.rs

```diff
@@ -139,6 +141,33 @@ impl RouterSuperServiceFactory for YamlRouterFactory {
         // Process the plugins.
         let plugins = create_plugins(&configuration, &schema, extra_plugins).await?;
 
+        let tls_root_store = configuration
+            .tls
+            .subgraphs
+            .as_ref()
+            .and_then(|subgraphs| subgraphs.certificate_authorities.as_ref())
+            .and_then(|certificate_authorities| {
+                let mut store = RootCertStore::empty();
+                let certificates = load_certs(certificate_authorities)
+                    .map_err(|e| {
+                        tracing::error!("could not parse certificate list: {e:?}");
+                    })
+                    .ok()?;
+                for certificate in certificates {
+                    store
+                        .add(&certificate)
+                        .map_err(|e| {
+                            tracing::error!("could not add certificate to root store: {e:?}");
+                        })
+                        .ok()?;
+                }
+                if store.is_empty() {
+                    None
```

Comment:
Per `subgraph_service.rs`, `None` here means "use native roots" while `Some` means "use *only* specified roots, leaving the native ones untrusted".

So specifying `certificate_authorities` makes native roots untrusted, except (through `is_empty` here) if loading fails in which case the native ones are used as fallback?

Reply:
maybe failing to load them should instead result in a hard error. If the subgraph uses a certificate from a custom CA, connections will fail with the native list

Reply:
I added an error in [4ae4fc6](https://github.com/apollographql/router/pull/2008/commits/4ae4fc6439bae80ef05458bbf837bd3ac93e739b)